aggregator/elem.go

```diff
@@ -130,10 +136,19 @@ type GaugeElem struct {
 	toConsume []timedGauge
 }
 
+func newElemBase(opts Options) elemBase {
+	return elemBase{
+		opts:    opts,
+		aggOpts: aggregation.NewOptions(),
+	}
+}
+
 // ResetSetData resets the counter and sets data.
-func (e *elemBase) ResetSetData(id id.RawID, policy policy.Policy) {
+func (e *elemBase) ResetSetData(id id.RawID, sp policy.StoragePolicy, aggTypes policy.AggregationTypes) {
```

Comment:
We can't directly use policy.Policy here, that would require decompress the policy.AggregationID to get the list of aggregation types